mender-convert-modify

```diff
@@ -95,30 +96,49 @@ mkdir -p work/rootfs
 sudo mount ${boot_part} work/boot
 sudo mount ${root_part} work/rootfs
 
-mkdir -p work/mender-deb/files
+mkdir -p work/deb-packages
 
 log_info "Installing Mender client and related files"
 
 deb_arch=$(probe_debian_arch_name)
-DEBIAN_REVISION="-1"
+
 if [ "${MENDER_CLIENT_VERSION}" = "latest" ]; then
-  DEBIAN_REVISION=""
+  deb_name=$(deb_from_repo_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "stable" "mender-client" "latest")
+elif [ "${MENDER_CLIENT_VERSION}" = "master" ]; then
+  deb_name=$(deb_from_repo_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "experimental" "mender-client" "latest")
+else
+  DEBIAN_REVISION="-1"
+  deb_name=$(deb_from_repo_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "experimental" "mender-client" "${MENDER_CLIENT_VERSION}${DEBIAN_REVISION}")
```

Comment:
Shouldn't this be `stable`? I assume the `else` block is when you specify an exact version.

Reply:
Thanks for asking, this might indicate that I should split the helper functions it two completely ones.

If you [look carefully at the code](https://github.com/mendersoftware/mender-convert/pull/263/commits/6fd2060b907af78b5abf4b955b9896e14e90e7e1#diff-3ef5430cd3b9a9759c354bc7fea58f7f0f49179cf317f5f3f5576d78a70a680fR52-R56), you can see that for specific version the distribution (i.e. stable or experimental) is ignored and the function get the deb package directly from the pool (where all packages are together, independently on the distribution).

As this caused a question by your side, do you think we should split `deb_from_repo_get` in two functions? The interface is somewhat already confusing with two optional parameters and one possibly ignored one.

Reply:
Hmm, you're right, this is confusing. Since the check inside `deb_from_repo_get` is `"${version}" = "latest"`, doesn't that mean that `stable` is the only one actually used?

I think I agree then, it would be better to split the function.

Reply:
Thanks. I will rework...

(the check inside the function for latest is, for example to get latest from experimental (which is master))

Reply:
@kacf I changed it now into two different functions. See last commit.